tests/common/models.py

```diff
@@ -122,10 +122,10 @@ class EvenSimplerMLP(torch.nn.Module):
 
     def __init__(self, num_features: int, device: str = 'cpu', num_out_features: int = 3):
         super().__init__()
-        fc1 = torch.nn.Linear(num_features, num_features, device=device, bias=False)
-        fc2 = torch.nn.Linear(num_features, num_out_features, device=device, bias=False)
+        self.fc1 = torch.nn.Linear(num_features, num_features, device=device, bias=False)
+        self.fc2 = torch.nn.Linear(num_features, num_out_features, device=device, bias=False)
```

Comment:
why attach to classes? this might be causing ur failure -- we intentionally removed it as it caused issues with FSDP

Reply:
To resolve this, I created `TPSimpleComposerMLP` which is just like `SimpleComposerMLP` but saves the each layer. Then I stopped saving the layer in `SimpleComposerMLP`.